src/openrct2/Editor.cpp

```diff
@@ -453,60 +453,54 @@ namespace Editor
      *
      *  rct2: 0x006AB9B8
      */
-    ObjectType CheckObjectSelection()
+    std::pair<ObjectType, rct_string_id> CheckObjectSelection()
     {
         bool isTrackDesignerManager = gScreenFlags & (SCREEN_FLAGS_TRACK_DESIGNER | SCREEN_FLAGS_TRACK_MANAGER);
 
         if (!isTrackDesignerManager)
         {
             if (!editor_check_object_group_at_least_one_selected(ObjectType::Paths))
             {
-                gGameCommandErrorText = STR_AT_LEAST_ONE_PATH_OBJECT_MUST_BE_SELECTED;
-                return ObjectType::Paths;
+                return { ObjectType::Paths, STR_AT_LEAST_ONE_PATH_OBJECT_MUST_BE_SELECTED };
             }
         }
 
         if (!editor_check_object_group_at_least_one_selected(ObjectType::Ride))
         {
-            gGameCommandErrorText = STR_AT_LEAST_ONE_RIDE_OBJECT_MUST_BE_SELECTED;
-            return ObjectType::Ride;
+            return { ObjectType::Ride, STR_AT_LEAST_ONE_RIDE_OBJECT_MUST_BE_SELECTED };
         }
 
         if (!isTrackDesignerManager)
         {
             if (!editor_check_object_group_at_least_one_selected(ObjectType::ParkEntrance))
             {
-                gGameCommandErrorText = STR_PARK_ENTRANCE_TYPE_MUST_BE_SELECTED;
-                return ObjectType::ParkEntrance;
+                return { ObjectType::ParkEntrance, STR_PARK_ENTRANCE_TYPE_MUST_BE_SELECTED };
             }
 
             if (!editor_check_object_group_at_least_one_selected(ObjectType::Water))
             {
-                gGameCommandErrorText = STR_WATER_TYPE_MUST_BE_SELECTED;
-                return ObjectType::Water;
+                return { ObjectType::Water, STR_WATER_TYPE_MUST_BE_SELECTED };
             }
         }
 
-        return ObjectType::None;
+        return { ObjectType::None, STR_NONE };
     }
 
     /**
      *
      *  rct2: 0x0066FEAC
      */
-    bool CheckPark()
+    std::pair<bool, rct_string_id> CheckPark()
```

Comment:
This one could be `std::optional<rct_string_id>` instead, as everytime it returns an actual error string, the `bool` is false

Reply:
Returning std::nullopt as "true" is a bit weird to me.

Reply:
I thought if it more like: "if it's empty, then there was no error"